inspirehep/modules/forms/validators/simple_fields.py

```diff
@@ -163,6 +163,13 @@ def pdf_validator(form, field):
     if field.data and not is_pdf_link(field.data):
         raise StopValidation(message)
 
+def year_validator(form, field):
+    """Validate that the year is not <1000 and >2050"""
+    message = 'Please, provide a valid year.'
+
+    if field.data and not 1000 <= int(field.data) <= 2050:
```

Comment:
`1000` and `2050` are magic numbers that should instead be derived from the schema, in particular: https://github.com/inspirehep/inspire-schemas/blob/2856417b4e3a0f5950e4482490b4648fff9f6a9d/inspire_schemas/records/hep.yml#L1259-L1265.